src/tests/functions/ConvertTo-PuppetResourceAPi.Tests.ps1

```diff
@@ -0,0 +1,122 @@
+Describe "ConvertTo-PuppetResourceApi" {
+  InModuleScope puppet.dsc {
+    Context "When Passed a DscResourceInfo object" {
+      Mock Get-DscResourceTypeInformation {
+        $DSCResource | Add-Member -MemberType NoteProperty -Name ParameterInfo -Value $DscResource.Name -PassThru
+      }
+      Mock Get-TypeContent     { return "Type: $($DscResource.Name)" }
+      Mock Get-ProviderContent { return "Provider: $($DscResource.Name)" }
+
+      $DscResources = [Microsoft.PowerShell.DesiredStateConfiguration.DscResourceInfo[]]@(
+        @{Name = 'FooResource'}
+        @{Name = 'BarResource'}
+      )
+      
+      $Result = $DscResources | ConvertTo-PuppetResourceApi
+      
+      It "processes once for each object in the pipeline" {
+        Assert-MockCalled Get-DscResourceTypeInformation -Times 2
+        $Result[0].Name | Should -BeExactly 'dsc_fooresource'
```

Comment:
should also test at least name for `$Result[1]` to make sure nothing untoward happened when iterating, same below on the other instances of this.